parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreBase.java

```diff
@@ -183,6 +186,9 @@ public void close() {
   @Override
   public void endRecord() {
     ++rowCount;
+    if (rowCount >= rowCountForNextRowCountCheck) {
```

Comment:
This is in a tight loop, so I think we should be a bit more careful. I think this check could go inside size check `if` statement if we guarantee that `rowCountForNextSizeCheck <= rowCountForNextRowCountCheck` by ensuring the size check count isn't larger than the row count check. Then we would only need to do one test for every row.